tests/test_sigmaJs.py

```diff
@@ -356,23 +356,18 @@ def test_get_def_id_by_rule_id():
 
     parser = graph.xml_parser.xml_parser(str(FIXTURE_DIR))
 
-    with pytest.raises(ValueError) as e:
-        parser.get_def_id_by_rule_id('hello')
-    assert str(
-        e.value) == 'err- 404 rule not found!'
+    with pytest.raises(Exception, match="err- 404 rule not found!"):
+        assert parser.get_def_id_by_rule_id('hello')
 
 
 def test_get_def_id_by_notselected_rule_id():
     src = 'test_data/ssg-fedora-ds-arf.xml'
 
     parser = tests.any_test_help.get_parser(src)
     rule_id = 'xccdf_org.ssgproject.content_rule_ntpd_specify_remote_server'
-    with pytest.raises(ValueError) as e:
-        parser.get_def_id_by_rule_id(rule_id)
-    assert str(
-        e.value) == ('err- rule "' +
-                     rule_id +
-                     '" was not selected, so there are no results.')
+
+    with pytest.raises(Exception, match="err- rule \"{}\" was not selected, so there are no results.".format(rule_id)):
```

Comment:
Very clever use of the `match` argument!
I have those small suggestions:


- Use single quote for quoting values, it is the common pattern. And try to avoid escaping quotes - you don't have to do it in Python (you can use `"`, `'`, or even `"""`, `'''` for the outer string to avoid quote escaping). This is probably an issue in the source code, not in the test.
- Make the matches more relaxed, so the tests won't break when you decide to make minor modifications to the error message. I would just check for the presence of the rule ID and the `not selected` phrase.